crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -59,29 +60,39 @@ def checks
 
           if Api::Crowbar.addons.include?("ceph")
             ceph_status = Api::Crowbar.ceph_status
-            ret[:ceph_healthy] = {
+            ret[:checks][:ceph_healthy] = {
               required: true,
               passed: ceph_status.empty?,
               errors: ceph_status.empty? ? {} : ceph_health_check_errors(ceph_status)
             }
           end
 
           ha_presence = Api::Pacemaker.ha_presence_check
-          ret[:ha_configured] = {
+          ret[:checks][:ha_configured] = {
             required: false,
             passed: ha_presence.empty?,
             errors: ha_presence.empty? ? {} : ha_presence_errors(ha_presence)
           }
 
           if Api::Crowbar.addons.include?("ha")
             clusters_health = Api::Pacemaker.health_report
-            ret[:clusters_healthy] = {
+            ret[:checks][:clusters_healthy] = {
               required: true,
               passed: clusters_health.empty?,
               errors: clusters_health.empty? ? {} : clusters_health_report_errors(clusters_health)
             }
           end
 
+          ret[:best_method] = if ret[:checks].any? { |_id, c| c[:required] && !c[:passed] }
+            "none"
+          elsif !ret[:checks].any? { |_id, c| (c[:required] || !c[:required]) && !c[:passed] }
+            "non-disruptive"
+          elsif !ret[:checks].any? do |_id, c|
+            (c[:required] && !c[:passed]) && (!c[:required] && c[:passed])
+          end
+            "disruptive"
+          end
+
           return ret unless upgrade_status.current_step == :prechecks
```

Comment:
Do we really need that new :checks submap for the `best_method` evaluation? For example here we're returning different `ret` than expected before...